Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserEntityInspectorWidget.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/AssetBrowser/AssetBrowserEntityInspectorWidget.h>
+#include <AzToolsFramework/AssetBrowser/Previewer/PreviewerFrame.h>
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/AssetBrowserModel.h>
+#include <AzToolsFramework/AssetBrowser/Entries/SourceAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/ProductAssetBrowserEntry.h>
+#include <QLayout>
+#include <AzCore/Asset/AssetTypeInfoBus.h>
+#include <AzCore/Asset/AssetManagerBus.h>
+
+namespace AzToolsFramework
+{
+    namespace AssetBrowser
+    {
+        AssetBrowserEntityInspectorWidget::AssetBrowserEntityInspectorWidget(QWidget* parent)
+        :QWidget(parent)
+        {
+            setLayout(new QVBoxLayout);
+            m_previewerFrame = new PreviewerFrame;
+            layout()->addWidget(m_previewerFrame);
+
+
+            AssetBrowserPreviewRequestBus::Handler::BusConnect();
+        }
+        AssetBrowserEntityInspectorWidget::~AssetBrowserEntityInspectorWidget()
+        {
+            AssetBrowserPreviewRequestBus::Handler::BusDisconnect();
+        }
+        void AssetBrowserEntityInspectorWidget::PreviewAsset(const AzToolsFramework::AssetBrowser::AssetBrowserEntry* selectedEntry)
+        {
+            m_previewerFrame->Display(selectedEntry);
```

Comment:
It seems that `PreviewerFrame::Display` doesn't do checks for null inside. The chain of calls that happens inside (`PreviewerRequests::GetPreviewerFactory` → `EditorCommonFeaturesSystemComponent::GetPreviewerFactory` → `SharedPreviewerFactory::IsEntrySupported`) may result in calling `GetThumbnailKey` on a null entry. I'd check for nullptr here, and clear if nullptr is indeed passed. Also I suggest documenting this behavior in the header file.